1.21.5 SCUBA Gear rework

EpyonProjects opened the pull request:
Main discussion and info source located on Discord in the [forum post](https://discord.com/channels/151141188961828864/1140750776617156628/1356855278519320707)

tl;dr:
- Removes the function clock, making SCUBA Helmet and Tank attributes.  
- Flippers is now an enchant that changes the attributes depending on whether the player is on land or in the water.
- Wetsuit is a new item for the remaining equipment slot: Legs.

The wetsuit allows you to float in water, or assist with swimming up- or downwards.  This enchant has a tick clock, making this rework the first introduction of GM4 enchants and the first case of an enchant clock.  The enchantment is displayed using lore because it uses multiple lines and displays the attributes manually.

EpyonProjects committed:
Initial commit

EpyonProjects committed:
Finished initial draft

EpyonProjects committed:
Update guidebook description to match readme

EpyonProjects committed:
Fixed texts and unified attributes
- updated helmet texture

misode commented:
@EpyonProjects Test needs to be fixed

EpyonProjects committed:
Fix recipe test

BPR02 commented on gm4_scuba_gear/data/gm4_scuba_gear/function/check_recipes.mcfunction:
```diff
@@ -1,4 +1,5 @@
+execute if score $crafted gm4_crafting matches 0 store success score $crafted gm4_crafting if score $slot_count gm4_crafting matches 8 if score $stack_size gm4_crafting matches 1 if data storage gm4_custom_crafters:temp/crafter {Items:[{Slot:0b,id:"minecraft:copper_ingot"},{Slot:2b,id:"minecraft:copper_ingot"},{Slot:3b,id:"minecraft:decorated_pot"},{Slot:4b,id:"minecraft:copper_ingot"},{Slot:5b,i ...
+execute if score $crafted gm4_crafting matches 0 store success score $crafted gm4_crafting if score $slot_count gm4_crafting matches 7 if score $stack_size gm4_crafting matches 1 if data storage gm4_custom_crafters:temp/crafter {Items:[{Slot:0b,id:"minecraft:dried_kelp"},{Slot:1b,id:"minecraft:copper_ingot"},{Slot:2b,id:"minecraft:dried_kelp"},{Slot:3b,id:"minecraft:dried_kelp"},{Slot:5b,id:"mine ...
+execute if score $crafted gm4_crafting matches 0 store success score $crafted gm4_crafting if score $slot_count gm4_crafting matches 4 if score $stack_size gm4_crafting matches 1 if data storage gm4_custom_crafters:temp/crafter {Items:[{Slot:0b,id:"minecraft:copper_ingot"},{Slot:2b,id:"minecraft:copper_ingot"},{Slot:3b,id:"minecraft:dried_kelp"},{Slot:5b,id:"minecraft:dried_kelp"}]} run loot repl ...
+execute if score $crafted gm4_crafting matches 0 store success score $crafted gm4_crafting if score $slot_count gm4_crafting matches 4 if score $stack_size gm4_crafting matches 1 if data storage gm4_custom_crafters:temp/crafter {Items:[{Slot:3b,id:"minecraft:copper_ingot"},{Slot:5b,id:"minecraft:copper_ingot"},{Slot:6b,id:"minecraft:dried_kelp"},{Slot:8b,id:"minecraft:dried_kelp"}]} run loot repl ...
```
I don't think this is necessary because Custom Crafters automatically shift the recipe to the top-left corner before checking recipes
```suggestion
```

BPR02 reviewed (changes requested):
Most stuff looks good. I left a comment about using scores instead of tags to track the buoyancy type since nbt checks are necessary to check tags in a predicate/enchantment, but it's probably not that big of a performance difference. 

There's also an inapplicable test left which needs to be removed.

EpyonProjects committed:
Oh my goodness, what is this, a commit?
- I should be asleep
- Addressed all of the concerns Beeps had
- Should have checked again at literally any point because this really was not much

EpyonProjects committed:
Actually address all of the changes Beeps had
- Before putting it onto public server

EpyonProjects committed:
Fix Scuba advancement for all gear

BPR02 committed:
Add custom armor assets (RP required)

Requires PR #1215

BPR02 commented:
https://github.com/EpyonProjects/GM4_Datapacks/pull/6 is a PR that adds custom assets to the armor models (i.e. shown on the player model). It however, requires a resource pack to work. This is to be discussed during next meeting.

BPR02 committed:
fix recipe tank trim

BPR02 reviewed (changes requested):
As noted in my last comment, the flippers do not give a speed boost while swimming.

BPR02 committed:
Use item_model instead of custom_model_data

This will make all items appear as missing textures. It's possible to make them look like leather armor without the RP, but making the item sprites missing textures should "force" the RP to be required. This is important because without the RP the armor is invisible, so we should highly discourage that by making the item textures missing without the RP. I.e. make it basically unplayable without the RP so people don't get invisible armor

BPR02 committed:
remove flippers of ostara

EpyonProjects committed:
Merge pull request #6 from BPR02/scuba-rp

Add custom armor assets (RP required)

runcows commented on gm4_boots_of_ostara/data/gm4_boots_of_ostara/item_modifier/boots_of_ostara.json:
```diff
@@ -51,47 +51,5 @@
-    {
-        "function": "minecraft:set_name",
-        "name": {
-            "translate": "item.gm4.flippers_of_ostara",
```
Make sure to remove this translation text from the boots of ostara translations file, since its not used anymore

misode commented:
@EpyonProjects Any progress on this? Apart from runcows' translation text comment, are there any issues left that need to be fixed?

Bloo-dev commented:
Converted to draft. Planned to be merged into the 26.1 update/dev branch once it is created.

EpyonProjects committed:
Swimming up with the wetsuit actually gives upward gravity instead of downward

EpyonProjects committed:
Merge branch 'master' of https://github.com/Gamemode4Dev/GM4_Datapacks into scuba-enchants

BPR02 reviewed (approved):
I'm not sure the upwards gravity with the wetsuit is actually doing anything, but the attribute seems to be working correctly now. Can confirm the flippers properly speed up the player while swimming.

misode commented on gm4_scuba_gear/data/minecraft/tags/enchantment/curse.json:
```diff
@@ -0,0 +1,7 @@
+  "replace": false,
+  "values": [
+    "gm4_scuba_gear:wetsuit",
+    "gm4_scuba_gear:flippers"
```
These values should be optional, because enchantments only reload when restarting the server, so it's possible that running a `/reload` could break the vanilla `curse` enchantment tag.

misode commented on gm4_scuba_gear/beet.yaml:
```diff
@@ -29,22 +29,9 @@ meta:
     credits:
```
This file should add a warning in the `notes` list that (1) it requires the resource pack, and (2) requires a server/world restart after adding the data pack.

Ideally we will be able to detect these two things as environment checks in the future.

misode commented on gm4_scuba_gear/data/gm4_scuba_gear/loot_table/item/flippers.json:
```diff
@@ -0,0 +1,80 @@
+{
```
Why were these loot tables moved from `items` to `item`? All our other modules use `items`, and it's also more consistent with vanilla's `blocks` and `chests`.

misode committed:
Fix review comments

misode committed:
Expand note

misode committed:
Remove old references to scuba_helmet player head
